rootio/tchain.go

```diff
@@ -0,0 +1,111 @@
+// Copyright 2018 The go-hep Authors.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package rootio
+
+type tchain struct {
+	trees []Tree
+}
+
+// Class returns the ROOT class of the argument.
+func (tchain) Class() string {
+	return "TChain"
+}
+
+// Name returns the name of the ROOT objet in the argument.
+func (t tchain) Name() string {
+	if len(t.trees) == 0 {
+		return ""
+	}
+	return t.trees[0].Name()
+}
+
+// Title returns the title of the ROOT object in the argument.
+func (t tchain) Title() string {
+	if len(t.trees) == 0 {
+		return ""
+	}
+	return t.trees[0].Title()
+}
+
+// Chain returns a tchain that is the concatenation of all the input Trees.
+func Chain(trees ...Tree) tchain {
+	var t tchain
```

Comment:
thinking a bit more about this, perhaps it would be better (memory-wise) to:
- check for the length of the `trees` slice and return a zero-value of `tchain` in case the length is 0
- create a `t.trees` slice with directly the correct length and then use the builtin `copy` function